File: bookshop-back-api/bookshop.webapi/Models/OrderFolder/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace bookshop.webapi.Models.OrderFolder
{
    public class Order
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public List<OrderItem> Items { get; set; }
        public AppUser User { get; set; }
        public float TotalPrice { get; set; }
        public string Address { get; set; }
        public string Country { get; set; }
        public string City { get; set; }
        public string PaymentId { get; set; }
    }
}

File: bookshop-front-angular/src/models/Order.ts
import { OrderItem } from "./OrderItem";

export class Order {
    public id: number = 0;
    public items: OrderItem[] = [];
    public totalPrice: number = 0;
    public address: string = "";
    public country: string = "";
    public city: string = "";
    public paymentId: string = "";
    public userName?: string = "";
    public email?: string = "";

    constructor(object: any) {
        this.id = object.id;
        this.items = object.items;
        this.totalPrice = object.totalPrice;
        this.address = object.address;
        this.country = object.country;
        this.city = object.city;
        this.paymentId = object.paymentId;
        this.email = object.user?.email;
        this.userName = object.user?.userName;
    }
}